domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java

```diff
@@ -193,6 +225,147 @@ static String message(Throwable t) {
     }
   }
 
+  static Future<Response> respond422(Method response422Method, String key, String value, String message) {
+    try {
+      Errors errors = ValidationHelper.createValidationErrorMessage(key, value, message);
+      Response response = (Response) response422Method.invoke(null, errors);
+      return Future.succeededFuture(response);
+    } catch (IllegalAccessException | InvocationTargetException e) {
+      throw new IllegalArgumentException(e);
+    }
+  }
+
+  /**
+   * Return the <code>respond422WithApplicationJson(Errors entity)</code> method.
+   * @param clazz class to search in
+   * @return the found method, or null if not found
+   */
+  static Method respond422method(Class<? extends ResponseDelegate> clazz) {
+    // this loop is 20 times faster than getMethod(...) if the method doesn't exist
+    // because it avoids the Exception that getMethod(...) throws.
+    for (Method method : clazz.getMethods()) {
+      if (method.getName().equals(RESPOND_422_WITH_APPLICATION_JSON)
+          && method.getParameterCount() == 1
+          && method.getParameters()[0].getType().equals(Errors.class)) {
+        return method;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Create a Response about the invalid uuid. Use clazz' respond422WithApplicationJson(Errors)
+   * if exists, otherwise use valueMethod.
+   * On exception create a Response using failResponseMethod(String exceptionMessage).
+   * If that also throws an exception create a failed future.
+   *
+   * <p>All exceptions are caught and reported via the returned Future.
+   */
+  static Future<Response> responseInvalidUuid(String field, String uuid,
+      Class<? extends ResponseDelegate> clazz, Method valueMethod, Method failResponseMethod) {
+
+    try {
+      Method respond422 = respond422method(clazz);
+      if (respond422 == null) {
+        return response(INVALID_UUID, valueMethod, failResponseMethod);
+      }
+      return respond422(respond422, field, uuid, INVALID_UUID);
+    } catch (Exception e) {
+      return response(e, failResponseMethod);
+    }
+  }
+
+  static Future<Response> responseForeignKeyViolation(String table, String id, PgExceptionFacade pgException,
+      Method response422method, Method valueMethod, Method failResponseMethod) {
+    try {
+      String detail = pgException.getDetail();
+      Matcher matcher = KEY_NOT_PRESENT_PATTERN.matcher(detail);
+      if (matcher.find()) {
+        String field = matcher.group(1);
+        String value = matcher.group(2);
+        String refTable = matcher.group(3);
+        String message = "Cannot set " + table + "." + field + " = " + value
+            + " because it does not exist in " + refTable + ".id.";
+        if (response422method == null) {
+          return response(message, valueMethod, failResponseMethod);
+        }
+        return respond422(response422method, table + "." + field, value, message);
+      }
+
```

Comment:
A lot of work for little gain ..

Reply:
The gain is small if only a single module will use this. However this is a library method to be used by many modules and many, many API endpoints.

Reply:
OK.. If you say so..